library/core/src/num/flt2dec/mod.rs

```diff
@@ -150,17 +150,13 @@ pub fn round_up(d: &mut [u8]) -> Option<u8> {
         Some(i) => {
             // d[i+1..n] is all nines
             d[i] += 1;
-            for j in i + 1..d.len() {
-                d[j] = b'0';
-            }
+            d.iter_mut().skip(i + 1).for_each(|c| *c = b'0');
             None
         }
-        None if d.len() > 0 => {
+        None if !d.is_empty() => {
```

Comment:
I do not find `!x.is_empty()` easier to read than `d.len() > 0`, frankly.

Reply:
I sympathize with that statement, but `is_empty` is in std, presumably because it makes it easier to express intent and thus be clearer and thus also easier to read. If the `not` is a problem, then this branch can always be switched with the next, I suppose.